services/121-service/src/registration/repositories/registration-scoped.repository.ts

```diff
@@ -111,9 +112,22 @@ export class RegistrationScopedRepository extends RegistrationScopedBaseReposito
     return this.repository.update(criteria, partialEntity);
   }
 
-  async getRegistrationByReferenceId(referenceId: string) {
+  public async getRegistrationByReferenceId({
+    referenceId,
+    programId,
+    relations = [],
+  }: {
+    referenceId: string;
+    programId?: number;
+    relations?: string[];
+  }) {
+    const whereOptions = {
+      referenceId: Equal(referenceId),
+      ...(programId != undefined ? { programId: Equal(programId) } : {}),
+    };
     return await this.repository.findOne({
-      where: { referenceId: referenceId },
+      where: whereOptions,
```

Comment:
I think putting the where options in-line here would make the code easier to read.